src/framework/ui/types/mnemonicstring.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2022 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "mnemonicstring.h"
+
+#ifndef NDEBUG
+#include "log.h"
+#endif
+
+using namespace mu::ui;
+
+static QString processMnemonics(const QString& str, bool show)
```

Comment:
showUnderline?

Reply:
It would be great to have some utests for MnemonicString since logic here isn't that trivial

Reply:
There can be only one mnemonic in the text. Should be processMnemonic

Also, you can see how qt guys solves a similar issue with mnemonic - QKeySequence::mnemonic

Reply:
Yes, I based my implementation on Qt's. However, in Qt this routine is implemented in at least 3 different places and in each place in a slightly different way, so I chose the "greatest common divisor" of all their implementations.